### PR TITLE
[FIX] website: Fix homepage and contactus menu URLs

### PR DESCRIPTION
**Pull request by yajo**
The `/page/homepage` menu must point to `/`, and menu items pointing to any view with the `website.` prefix must remove that.

@Tecnativa

### CODE REVIEW

File: addons/website/migrations/11.0.1.0/post-migration.py [hunk 1]
```diff
@@ -68,6 +68,23 @@ def add_website_homepages(env):
...
+    """Update some menu URLs that must change."""
+    # Fix menus that include the "website." prefix
+    menus = env["website.menu"].search([
+        ("url", "=like", "/page/website.%"),
+    ])
+    for menu in menus:
+        menu.url = menu.url.replace("website.", "", 1)
+    # Fix menus pointing to "/page/homepage"
+    menus = env["website.menu"].search([
+        ("url", "=", "/page/homepage"),
```
Comment (pedrobaeza) [inline addons/website/migrations/11.0.1.0/post-migration.py]: This must use `=like` the same as above as you can have anchors to homepage (`/page/homepage#anchor`).

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: addons/website/migrations/11.0.1.0/post-migration.py (hunk 1 above)
Comment (pedrobaeza) [reply addons/website/migrations/11.0.1.0/post-migration.py]: As talked through chat, we need to split this is 2 conditions.

Comment (yajo) [reply addons/website/migrations/11.0.1.0/post-migration.py]: Fixed, thanks!

File: addons/website/migrations/11.0.1.0/post-migration.py [hunk 2]
```diff
@@ -68,6 +68,28 @@ def add_website_homepages(env):
...
+    ])
+    for menu in menus:
+        menu.url = menu.url.replace("website.", "", 1)
+    # Fix menus pointing to "/page/homepage"
+    menus = env["website.menu"].search([
+        "|", "|", "|", "|", "|",
+        ("url", "=", "/page/homepage"),
+        ("url", "=", "/page/homepage/"),
+        ("url", "=like", "/page/homepage?%"),
+        ("url", "=like", "/page/homepage/?%"),
```
Comment (pedrobaeza) [inline addons/website/migrations/11.0.1.0/post-migration.py]: For this case you will have as result `//?%`

File: addons/website/migrations/11.0.1.0/post-migration.py [hunk 3]
```diff
@@ -68,6 +68,28 @@ def add_website_homepages(env):
...
+        ("url", "=like", "/page/homepage/#%"),
```
Comment (pedrobaeza) [inline addons/website/migrations/11.0.1.0/post-migration.py]: For this case you will have as result `//#%`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: addons/website/migrations/11.0.1.0/post-migration.py (hunk 2 above)
Comment (yajo) [reply addons/website/migrations/11.0.1.0/post-migration.py]: Odoo always autoremoves the trailing `/` (no matter if the controller exists or not) so there is zero chances that a user copies and pastes this link there, and when using autofill Odoo doesn't provide answers that end with `/`.

Taking that into account, I just removed the searches that finished in `/`. There's really so low chance those exist, and even more in a page such as homepage, which is generated by XML data.

File: addons/website/migrations/11.0.1.0/post-migration.py (hunk 3 above)
Comment (yajo) [reply addons/website/migrations/11.0.1.0/post-migration.py]: Fixed also, see above.

**Comment by yajo** [comment]
With previous implementation there were the problems discussed about trailing slashes, and still the links in page bodies would fail.

After all, it's simpler and more effective to just add a 301 redirection, which I added and did the same for contactus controller, which has been moved also.

This looks better now, honestly. Ready to merge IMHO.

**Pull request closed without merging**